src/frequenz/sdk/timeseries/logical_meter/_formula_steps.py

```diff
@@ -265,7 +254,12 @@ def apply(self, eval_stack: List[Optional[float]]) -> None:
         """
         if self._next_value is None:
             raise RuntimeError("No next value available to append.")
-        if self._next_value.value is None and self._nones_are_zeros:
-            eval_stack.append(0.0)
+
+        next_value = self._next_value.value
+        if next_value is None or isnan(next_value) or isinf(next_value):
+            if self._nones_are_zeros:
+                eval_stack.append(0.0)
+            else:
+                eval_stack.append(float("NaN"))
```

Comment:
NB: Python >= 3.10 introduces `math.nan` for symbolic access to `NaN`.

Reply:
Then it can/should be updated once 3.8 and 3.9 are no longer supported